### DISCUSSION TITLE
Anyone successfully deployed this in AWS Govcloud? Looking into deploying this using the ECS feature.

### DISCUSSION THREAD

**Discussion by 404GoatFound** [NUWCDIVNPT/stig-manager#1422] [General]
As said, i'm having issues deploying this in aws govcloud. I can't use a docker-compose.yml so I have to enter all variables manually. Looking to speak with someone that's has success as we've had many issues with the keycloak and stig managerconfig.

**Comment by 404GoatFound** [comment]
Update, I've figured it out. If anyone is having issues feel free to reach out.

**Reply by cd-rite** [reply]
@Bwall95 That's great! We don't have a lot of experience with AWS directly, so we can't offer too much help there with specifics. What turned out to be the key that got you working?

**Reply by 404GoatFound** [reply]
> @Bwall95 That's great! We don't have a lot of experience with AWS directly, so we can't offer too much help there with specifics. What turned out to be the key that got you working?

There were a lot of configuration that went into it. I may not have the best solution but it worked. 

The main issue was accessing everything from a non localhost machine. I'm working on setting up an Elastic Load Balancer in AWS to help with some of my issues. 

I ended up changing all environment variables to be the IP of the host machine instead of "localhost". I make a DNS entry in windows to point to the host IP, which NGINX is listening on. Also the newest DOD certs need downloaded from cyber.mil and converted to .pem format for the reverse proxy to use. I had to get a DOD issued certificate for the NGINX server to use, along with a .key file for the cert. I had to run a docker command that disabled the SSL requirement for the master realm so I could initially sign into the keycloak realm. 

Also, configuring a task definition in ECS is similar to docker compose, but much harder. I can share specific variables if anyone wants to reach out to this post with issues in AWS.

**Comment by rlmass** [comment]
Would you be willing to write up a guide to doing this? This may be very helpful for others!

**Reply by 404GoatFound** [reply]
So, we ended up moving to an EKS cluster instead of ECS. It's much easier. If you wanted to utilize that route I can assist. I deployed the EKS cluster using Terraform and could share that code to make it an extremely easy process for you.

**Reply by petlaugh-33** [reply]
Yeah thatd be very helpful!
